core/src/test/java/com/coderanch/blackjack/CardsTest.java

```diff
@@ -123,4 +123,35 @@ public void getStandardDeck_returnsUnmodifiableSet() {
     public void getStandardDeck_returnsAllCombinationsOfRankAndSuit() {
         assertThat(Cards.getStandardDeck().size(), is(equalTo(Rank.values().length * Suit.values().length)));
     }
+
+    /**
+     * Tests that {@link Cards#getShuffledStandardDeck(Random)} returns a set that contains all possible distinct cards that can be
+     * made using a combination of a {@link Rank} and a {@link Suit}.
+     */
+    @Theory
+    public void getShuffledStandardDeck_returnsAllCombinationsOfRankAndSuit() {
+        var random = new FixedRandom(List.of(5,4,2,1,3));
+        assertThat("Check that the card count is correct after shuffling", Cards.getShuffledStandardDeck(random).size(), is(equalTo(Rank.values().length * Suit.values().length)));
+    }
+
+    private static class FixedRandom extends Random {
```

Comment:
Good job. There are a couple of issues here, but I want to make some general refactors in a separate pull request. This is fine for now.

Reply:
thanks